File: app/src/main/java/com/example/mutidemo/mvvm/retrofit/RetrofitServiceManager.java
package com.example.mutidemo.mvvm.retrofit;


import com.example.mutidemo.util.DemoConstant;

import okhttp3.ResponseBody;
import rx.Observable;


public class RetrofitServiceManager {

    private static final RetrofitService api = RetrofitFactory.createRetrofit(DemoConstant.BASE_URL, RetrofitService.class);

    public static Observable<ResponseBody> obtainWeatherData(String city, int cityId, int cityCode) {
        return api.obtainWeatherData(DemoConstant.APP_KEY, city, cityId, cityCode);
    }
}

File: app/src/main/java/com/example/mutidemo/mvvm/retrofit/RetrofitServiceManager.kt
package com.example.mutidemo.mvvm.retrofit

import com.example.mutidemo.mvvm.model.WeatherModel
import com.example.mutidemo.util.DemoConstant

object RetrofitServiceManager {

    private val api by lazy { RetrofitFactory.createRetrofit(RetrofitService::class.java) }

    /**
     * 获取巡检详情
     */
    suspend fun obtainWeatherData(city: String, cityId: Int, cityCode: Int): WeatherModel {
        return api.obtainWeatherData(
            appkey = DemoConstant.APP_KEY,
            city = city,
            cityid = cityId,
            citycode = cityCode
        )
    }
}